### PR TITLE
Add frozen_string_literal pragma to prevent duplicate string allocations

### PR DESCRIPTION
**Pull request by oniofchaos**
Only thing I couldn't figure out was how to properly get the freeze in the `MAGIC` nested arrays.

### CODE REVIEW

**Comment by oniofchaos** [comment]
@minad anything I can do to help get this merged?

File: script/generate-mime.rb [hunk 1]
```diff
@@ -139,24 +140,24 @@ def get_matches(parent)
 puts "  # :nodoc:"
 puts "  EXTENSIONS = {"
 extensions.keys.sort.each do |key|
-  puts "    '#{key}' => '#{extensions[key]}',"
+  puts "    '#{key}' => '#{extensions[key]}'.freeze,"
```
Comment (minad) [inline script/generate-mime.rb]: why not just put the pragma in the beginning?

**Comment by minad** [comment]
> Only thing I couldn't figure out was how to properly get the freeze in the MAGIC nested arrays.

Did you find a solution? There shouldn't be those many freeze calls in the table.

File: script/generate-mime.rb (hunk 1 above)
Comment (oniofchaos) [reply script/generate-mime.rb]: I'm not sure what you mean :(

**Comment by oniofchaos** [comment]
Unfortunately no I wasn't able to

**Comment by oniofchaos** [comment]
Making a new PR since I don't have access to the original fork anymore.

**Pull request closed without merging**